lib/ui/blocs/deeplink/deeplink_bloc.dart

```diff
@@ -1,34 +1,76 @@
 import 'dart:async';
 
+import 'package:app_links/app_links.dart';
 import 'package:bloc/bloc.dart';
 import 'package:firebase_dynamic_links/firebase_dynamic_links.dart';
 import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:get/get.dart';
+import 'package:hypha_wallet/core/crypto/dart_esr/dart_esr.dart';
+import 'package:hypha_wallet/core/crypto/seeds_esr/scan_qr_code_result_data.dart';
 import 'package:hypha_wallet/core/logging/log_helper.dart';
+import 'package:hypha_wallet/ui/home_page/usecases/parse_qr_code_use_case.dart';
 
 part 'deeplink_bloc.freezed.dart';
 part 'deeplink_event.dart';
 part 'deeplink_state.dart';
 part 'page_command.dart';
 
 class DeeplinkBloc extends Bloc<DeeplinkEvent, DeeplinkState> {
-  DeeplinkBloc() : super(const DeeplinkState()) {
+  final ParseQRCodeUseCase _parseQRCodeUseCase;
+  final _appLinks = AppLinks();
+
+  StreamSubscription<String>? _linkSubscription;
+
+  DeeplinkBloc(this._parseQRCodeUseCase) : super(const DeeplinkState()) {
     initDynamicLinks();
 
     on<_IncomingFirebaseDeepLink>(_incomingFirebaseDeepLink);
+    on<_IncomingESRLink>(_incomingESRLink);
     on<_ClearPageCommand>((_, emit) => emit(state.copyWith(command: null)));
   }
 
+  void _debugShowMessage(String message) {
+    Get.showSnackbar(GetSnackBar(message: message, duration: const Duration(seconds: 5)));
+  }
+
   Future<void> initDynamicLinks() async {
-    final PendingDynamicLinkData? initialLink = await FirebaseDynamicLinks.instance.getInitialLink();
+    // Handle initial firebase links - such as post install
+    final PendingDynamicLinkData? initialDeepLink = await FirebaseDynamicLinks.instance.getInitialLink();
+
+    if (initialDeepLink != null) {
+      LogHelper.d('initial deep link: ${initialDeepLink.link}');
+      if (!initialDeepLink.link.isEsr()) {
+        add(DeeplinkEvent.incomingFirebaseDeepLink(initialDeepLink.link));
+      }
+    }
 
-    if (initialLink != null) {
-      LogHelper.d('initial link: ${initialLink.link}');
-      add(DeeplinkEvent.incomingFirebaseDeepLink(initialLink.link));
+    // Handle initial ESR links
+    final initialUriString = await _appLinks.getInitialAppLinkString();

```

Comment:
now that i know what app links is., 

Can we move the use of this AppLinks package into a service. Similar to Push notification service. 

If we need this PR ASAP, we can do on a diff PR.

Reply:
Not sure how to do that, it's a very small piece of code...